mysql-test/suite/rpl/t/rpl_extra_flags_event_len_inconsistent.test

```diff
@@ -0,0 +1,74 @@
+#
+#   This test enusres that, when a GTID event is constructed by reading its
+# content from a binlog file, the reader (e.g. replica, in this test) cannot
+# read beyond the length of the GTID event, if the extra_flags is not
+# consistent with the format of the event itself.
+#
+#   To test this, we create a transaction with its length > GTID_HEADER_LEN
+# (otherwise, it is zero padded and terminated early). To extend its length,
+# we give the transaction a commit id.
```

Comment:
"we create a transaction" - that was hard to understand, maybe "we create a GTID event".